src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
-    DDIX_FILTER_SEPARATOR,
-    DDIX_FILTER_SELECTED,
-    DDIX_FILTER_NONSELECTED,
-};
 
-struct ObjectListItem
-{
-    const ObjectRepositoryItem* repositoryItem;
-    std::unique_ptr<rct_object_filters> filter;
-    uint8_t* flags;
-};
+public:
 
-static rct_string_id GetRideTypeStringId(const ObjectRepositoryItem* item);
+    /**
+     *
+     *  rct2: 0x006AA64E
+     */
+    void OnOpen() override
+    {
+        auto& targetWidgets = _window_editor_object_selection_widgets;
+        if (!_window_editor_object_selection_widgets_initialised)
+        {
+            _window_editor_object_selection_widgets_initialised = true;
+            auto tabWidget = targetWidgets[targetWidgets.size() - 2];
+            for (size_t i = 1; i < std::size(ObjectSelectionPages); i++)
+            {
+                targetWidgets.insert(targetWidgets.end() - 1, tabWidget);
+            }
+        }
 
-using sortFunc_t = bool (*)(const ObjectListItem&, const ObjectListItem&);
+        widgets = targetWidgets.data();
 
-static std::vector<ObjectListItem> _listItems;
-static int32_t _listSortType = RIDE_SORT_TYPE;
-static bool _listSortDescending = false;
-static std::unique_ptr<Object> _loadedObject;
+        sub_6AB211();
+        reset_selected_object_count_and_size();
 
-static void VisibleListDispose()
-{
-    _listItems.clear();
-    _listItems.shrink_to_fit();
-}
+        enabled_widgets = (1ULL << WIDX_ADVANCED) | (1ULL << WIDX_INSTALL_TRACK) | (1ULL << WIDX_FILTER_DROPDOWN)
+            | (1ULL << WIDX_FILTER_TEXT_BOX) | (1ULL << WIDX_FILTER_CLEAR_BUTTON) | (1ULL << WIDX_CLOSE)
+            | (1ULL << WIDX_LIST_SORT_TYPE) | (1UL << WIDX_LIST_SORT_RIDE);
 
-static bool VisibleListSortRideName(const ObjectListItem& a, const ObjectListItem& b)
-{
-    auto nameA = a.repositoryItem->Name.c_str();
-    auto nameB = b.repositoryItem->Name.c_str();
-    return strcmp(nameA, nameB) < 0;
-}
+        widgets[WIDX_FILTER_TEXT_BOX].string = _filter_string;
 
-static bool VisibleListSortRideType(const ObjectListItem& a, const ObjectListItem& b)
-{
-    auto rideTypeA = language_get_string(GetRideTypeStringId(a.repositoryItem));
-    auto rideTypeB = language_get_string(GetRideTypeStringId(b.repositoryItem));
-    int32_t result = String::Compare(rideTypeA, rideTypeB);
-    return result != 0 ? result < 0 : VisibleListSortRideName(a, b);
-}
+        _filter_flags = gConfigInterface.object_selection_filter_flags;
+        std::fill_n(_filter_string, sizeof(_filter_string), 0x00);
 
-static void VisibleListRefresh(rct_window* w)
-{
-    int32_t numObjects = static_cast<int32_t>(object_repository_get_items_count());
+        for (size_t i = WIDX_TAB_1; i < WIDX_TAB_1 + std::size(ObjectSelectionPages); i++)
+        {
+            enabled_widgets |= (1LL << i);
+        }
+        WindowInitScrollWidgets(this);
 
-    VisibleListDispose();
-    w->selected_list_item = -1;
+        selected_tab = 0;
+        selected_list_item = -1;
+        min_width = 755;
```

Comment:
use the WW and WH names instead of the magic numbers.

Reply:
https://github.com/OpenRCT2/OpenRCT2/commit/0460b8287005ce74f16eb1cfbf49a2c3ae80272c